### PR TITLE
bump: bump up dependencies

### PR DESCRIPTION
**Pull request by Two-Hearts**
This PR is for `release-1.2` branch.

### CODE REVIEW

**Review by JeyJeyGao — APPROVED** [review]
LGTM

File: test/e2e/go.mod [hunk 1]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/onsi/ginkgo/v2 v2.11.0
 	github.com/onsi/gomega v1.27.10
 	github.com/opencontainers/image-spec v1.1.0
-	oras.land/oras-go/v2 v2.4.0
```
Comment (shizhMSFT) [inline test/e2e/go.mod:10]: Can you also send another PR to update `.github/dependabot.yml` to cover `/test/e2e/`?

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: test/e2e/go.mod (hunk 1 above)
Comment (Two-Hearts) [reply test/e2e/go.mod:10]: That's a good idea. Will do.

Comment (Two-Hearts) [reply test/e2e/go.mod:10]: PR created: https://github.com/notaryproject/notation/pull/1030

**Review by priteshbandi — APPROVED** [review]
LGTM
